test/edu/stanford/dlss/wowza/TestSulWowza.java

```diff
@@ -733,6 +735,45 @@ public void verifyTokenAgainstStacksService_validUrl()
     }
 
     @Test
+    public void verifyTokenAgainstStacksService_filenameNeedsEncoding()
+    {
+        WMSProperties mockProperties = mock(WMSProperties.class);
+        when(mockProperties.getPropertyStr(anyString(), anyString())).thenReturn(SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL);
+        IApplicationInstance appInstanceMock = mock(IApplicationInstance.class);
+        when(appInstanceMock.getProperties()).thenReturn(mockProperties);
+        testModule.onAppStart(appInstanceMock);
+        String druid = "oo000oo0000";
+        String filename = "file name with spaces; and an unusual char @ then end.ext!";
```

Comment:
suggestion taken:
https://github.com/sul-dlss/dlss-wowza/pull/18/files#diff-c544eacecdbf5b2f71f9ab070fb052b2R749
https://github.com/sul-dlss/dlss-wowza/pull/18/files#diff-c544eacecdbf5b2f71f9ab070fb052b2R771

those should be the same strings, and they should include all the characters you mentioned, all the weird ones that were already there, and some others.  i was able to use that string to create a file on both my mac and the wowza VM (i had to backslash escape the backtick in the string on the command line, but otherwise, same).